### PR TITLE
Harvest basic borrow functionality

### PR DESCRIPTION
**Pull request by denalimarsh**
This PR contains basic harvest functionality.

General borrow steps:
```bash
kvcli tx pricefeed postprice usdx 100 99999999999 --from [oracle-addr]
kvcli tx harvest deposit 1000000usdx lp --from [usdx-holder-addr]
kvcli tx harvest borrow 1000000usdx --from [usdx-holder-addr]
```

My local borrow steps to successfully deposit USDX and borrow KAVA:
```
# Set asset prices in price feed
kvcli tx pricefeed postprice usdx:usd 1 99999999999 --from validator
kvcli tx pricefeed postprice kava:usd 10 99999999999 --from validator
kvcli tx pricefeed postprice bnb:usd 100 99999999999 --from validator
kvcli tx pricefeed postprice bnb:usd:30 100 99999999999 --from validator

# Load USDX onto testing account by creating a CDP
kvcli tx cdp create 1000000000bnb 100000000usdx bnb-a --from kava1g0qywkx6mt5jmvefv6hs7c7h333qas5ks63a6t --gas 500000

# Load KAVA into Harvest module account (required in order to borrow from Harvest module account)
kvcli tx harvest deposit 10000000ukava lp --from kava1g0qywkx6mt5jmvefv6hs7c7h333qas5ks63a6t

# Deposit USDX and borrow KAVA
kvcli tx harvest deposit 10000000usdx lp --from kava1g0qywkx6mt5jmvefv6hs7c7h333qas5ks63a6t
kvcli tx harvest borrow 1000000ukava --from kava1g0qywkx6mt5jmvefv6hs7c7h333qas5ks63a6t
```

### CODE REVIEW

File: x/harvest/keeper/keeper.go [hunk 1]
```diff
@@ -178,7 +182,70 @@ func (k Keeper) IterateClaimsByTypeAndDenom(ctx sdk.Context, depositType types.D
 	}
 }
 
+// GetDepositsByUser gets all deposits for an individual user
+func (k Keeper) GetDepositsByUser(ctx sdk.Context, user sdk.AccAddress) []types.Deposit {
```
Comment (karzak) [inline x/harvest/keeper/keeper.go:186]: If we keep the architecture this way - `GetDepositsByUser` can be improved by iterating the denoms (params.MoneyMarkets[x].Denom) and calling `GetDeposit(ctx,user,denom, types.lp)` for each denom.

**Review by karzak — CHANGES_REQUESTED** [review]
Overall, outline looks good. Left some thoughts on params and how we're storing deposits/withdraws.

File: x/harvest/types/params.go [hunk 2]
```diff
@@ -28,6 +30,7 @@ type Params struct {
 	Active                         bool                           `json:"active" yaml:"active"`
 	LiquidityProviderSchedules     DistributionSchedules          `json:"liquidity_provider_schedules" yaml:"liquidity_provider_schedules"`
 	DelegatorDistributionSchedules DelegatorDistributionSchedules `json:"delegator_distribution_schedules" yaml:"delegator_distribution_schedules"`
+	BorrowLimits                   BorrowLimits                   `json:"borrow_limits" yaml:"borrow_limits"`
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/harvest/types/params.go:33]: Are you envisioning that `BorrowLimit` includes both the asset's LoanToValue percentage and an absolute borrow limit (similar to the CDP collateral type limit)?

[the remark this answers is not in the document]
Comment (karzak) [reply x/harvest/types/params.go:33]: Yeah, something like

```
type BorrowLimit struct {
  LoanToValue sdk.Dec
  Unlimited bool // if false, there is an arbitrary limit on borrows. if true, borrow is only limited by supply
  Limit sdk.Int // if Unlimited is false, this is the absolute borrow limit, regardless of supply
}

[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/harvest/types/params.go]: Added in https://github.com/Kava-Labs/kava/issues

[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/harvest/types/params.go]: Added in https://github.com/Kava-Labs/kava/pull/702/commits/9c1599c0cdea94fd8ccdfd693fb369bb7751d69d

File: x/pricefeed/keeper/params.go [hunk 3]
```diff
@@ -58,5 +60,19 @@ func (k Keeper) GetMarket(ctx sdk.Context, marketID string) (types.Market, bool)
 		}
 	}
 	return types.Market{}, false
+}
 
+// GetLiveMarketIDByDenom gets the live market ID by denom, selecting 'bnb:usd' over 'bnb:usd:30' for input 'bnb'
+func (k Keeper) GetLiveMarketIDByDenom(ctx sdk.Context, denom string) (string, bool) {
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/pricefeed/keeper/params.go]: Added in https://github.com/Kava-Labs/kava/pull/702/commits/bc0e32f0f80dd4d5b4f5afdb73bda2aaf11d85bd

File: x/harvest/types/borrow.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Borrow defines an amount of coins borrowed from a harvest module account
+type Borrow struct {
+	Borrower sdk.AccAddress `json:"borrower" yaml:"borrower"`
+	Amount   sdk.Coin       `json:"amount" yaml:"amount"`
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/harvest/types/borrow.go:10]: Created issue https://github.com/Kava-Labs/kava/issues/704 for this

File: x/harvest/keeper/keeper.go (hunk 1 above)
Comment (denalimarsh) [reply x/harvest/keeper/keeper.go:186]: Created issue https://github.com/Kava-Labs/kava/issues/705 for this

File: x/harvest/types/params.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/harvest/types/params.go]: Ah, just saw your latest comment. I implemented this as a required `MaximumLimit` - I'll change it during the review process or during future iterations.

File: x/harvest/keeper/keeper.go [hunk 5]
```diff
@@ -178,7 +182,70 @@ func (k Keeper) IterateClaimsByTypeAndDenom(ctx sdk.Context, depositType types.D
...
+		var borrow types.Borrow
+		k.cdc.MustUnmarshalBinaryBare(iterator.Value(), &borrow)
+		if cb(borrow) {
+			break
+		}
+	}
+}
+
+// GetBorrowsByUser gets all borrows for an individual user
+func (k Keeper) GetBorrowsByUser(ctx sdk.Context, user sdk.AccAddress) []types.Borrow {
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/harvest/keeper/keeper.go:242]: Encompassed by issues https://github.com/Kava-Labs/kava/issues/704, https://github.com/Kava-Labs/kava/issues/705

File: x/harvest/keeper/borrow.go [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+	return nil
+}
+
+func (k Keeper) validateBorrowUser(ctx sdk.Context, borrower sdk.AccAddress, amount sdk.Coin) error {
+	var borrowMoneyMarket types.MoneyMarket
+	var borrowAssetPrice sdk.Dec
+	var borrowUSDValue sdk.Dec
+	if amount.Denom == USDX {
+		borrowUSDValue = sdk.NewDecFromInt(amount.Amount).Quo(sdk.NewDec(int64(math.Pow(10, 6)))) // TODO: include conversion_factor in params
```
Comment (karzak) [inline x/harvest/keeper/borrow.go:67]: Resolve this todo before merge - code is too hard to follow with these USDX specific if/else statements

File: x/harvest/types/params.go [hunk 7]
```diff
@@ -218,26 +221,97 @@ func (ds DistributionSchedule) GetMultiplier(name MultiplierName) (Multiplier, b
...
+		return fmt.Errorf("loan-to-value cannot be greater than 100: %s", bl.LoanToValue)
+	}
+	return nil
+}
+
+// MoneyMarket is a money market for an individual asset
+type MoneyMarket struct {
+	Denom        string      `json:"denom" yaml:"denom"`
+	BorrowLimit  BorrowLimit `json:"borrow_limit" yaml:"borrow_limit"`
+	SpotMarketID string      `json:"spot_market_id" yaml:"spot_market_id"`
```
Comment (karzak) [inline x/harvest/types/params.go:256]: Add `ConversionFactor`

File: x/harvest/keeper/borrow.go [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+	if borrowUSDValue.GT(borrowValueLimit) {
+		return sdkerrors.Wrapf(types.ErrInsufficientLoanToValue,
+			"requested borrow %s is greater than maximum valid borrow %s",
+			amount, sdk.NewCoin(amount.Denom, borrowValueLimit.Quo(borrowAssetPrice).TruncateInt()))
+	}
+
+	return nil
+}
+
+func (k Keeper) validateBorrowSystem(ctx sdk.Context, borrower sdk.AccAddress, amount sdk.Coin) error {
```
Comment (karzak) [inline x/harvest/keeper/borrow.go:135]: delete this function

File: x/harvest/keeper/borrow.go [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+	}
+
+	if err := k.validateBorrowSystem(ctx, borrower, amount); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (k Keeper) validateBorrowUser(ctx sdk.Context, borrower sdk.AccAddress, amount sdk.Coin) error {
```
Comment (karzak) [inline x/harvest/keeper/borrow.go:62]: add a comment that describes what this does - I would change name as 'BorrowUser' doesn't seem descriptive. Maybe `validateLoanToValue`?

File: x/harvest/keeper/borrow.go [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+		locBorrowAssetPrice, err := k.pricefeedKeeper.GetCurrentPrice(ctx, locBorrowMoneyMarket.SpotMarketID)
+		if err != nil {
+			return sdkerrors.Wrapf(types.ErrPriceNotFound, "1: no price found for market %s", locBorrowMoneyMarket.SpotMarketID)
+		}
+		borrowMoneyMarket = locBorrowMoneyMarket
+		borrowAssetPrice = locBorrowAssetPrice.Price
+		borrowUSDValue = sdk.NewDecFromInt(amount.Amount).Mul(borrowAssetPrice)
+	}
+
+	// Get the user's deposits
```
Comment (karzak) [inline x/harvest/keeper/borrow.go:82]: move the `len(deposits) == 0` check to before checking USD value of the borrow.

File: x/harvest/keeper/borrow.go [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+	deposits := k.GetDepositsByUser(ctx, borrower)
+	if len(deposits) == 0 {
+		return sdkerrors.Wrapf(types.ErrDepositsNotFound, "no deposits found for %s", borrower)
+	}
+
+	// Get the total value of the user's deposits
+	totalValueDeposits := sdk.ZeroDec()
```
Comment (karzak) [inline x/harvest/keeper/borrow.go:89]:
```suggestion
	totalUSDValueDeposits := sdk.ZeroDec()
```

File: x/harvest/keeper/borrow.go [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+	totalValueBorrows := sdk.ZeroDec()
+	for _, borrow := range borrows {
+		if borrow.Amount.Denom == USDX {
+			totalValueBorrows = totalValueBorrows.Add(sdk.NewDecFromInt(borrow.Amount.Amount))
+		} else {
+			borrowAssetPrice, err := k.pricefeedKeeper.GetCurrentPrice(ctx, borrowMoneyMarket.SpotMarketID)
+			if err != nil {
+				return err
+			}
+			borrowUSDValue := sdk.NewDecFromInt(borrow.Amount.Amount).Mul(borrowAssetPrice.Price)
```
Comment (karzak) [inline x/harvest/keeper/borrow.go:118]: This will overwrite the previous value of borrowUSDValue, make the comparison on line 126 invalid.

File: x/harvest/keeper/borrow.go [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+		return err
+	}
+
+	return nil
+}
+
+func (k Keeper) validateBorrowUser(ctx sdk.Context, borrower sdk.AccAddress, amount sdk.Coin) error {
+	var borrowMoneyMarket types.MoneyMarket
+	var borrowAssetPrice sdk.Dec
+	var borrowUSDValue sdk.Dec
```
Comment (karzak) [inline x/harvest/keeper/borrow.go:65]:
```suggestion
	var proprosedBorrowUSDValue sdk.Dec
```

File: x/harvest/keeper/borrow.go [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+				return sdkerrors.Wrapf(types.ErrPriceNotFound, "2: no price found for market %s", depositMoneyMarket)
+			}
+			depositUSDValue := sdk.NewDecFromInt(deposit.Amount.Amount).Mul(depositAssetPrice.Price)
+			totalValueDeposits = totalValueDeposits.Add(depositUSDValue)
+		}
+	}
+
+	// Get the total value of the user's borrows
+	borrows := k.GetBorrowsByUser(ctx, borrower)
+	totalValueBorrows := sdk.ZeroDec()
```
Comment (karzak) [inline x/harvest/keeper/borrow.go:109]:
```suggestion
	totalUSDValuePreviousBorrows := sdk.ZeroDec()
```

File: x/harvest/keeper/params.go [hunk 15]
```diff
@@ -37,3 +39,14 @@ func (k Keeper) GetDelegatorSchedule(ctx sdk.Context, denom string) (types.Deleg
 	}
 	return types.DelegatorDistributionSchedule{}, false
 }
+
+// GetMoneyMarket returns the corresponding Money Market param for a specific denom
+func (k Keeper) GetMoneyMarket(ctx sdk.Context, denom string) (types.MoneyMarket, bool) {
```
Comment (karzak) [inline x/harvest/keeper/params.go:44]: created #706  - would like to avoid calling `GetParams` when handling a `MsgBorrow`

File: x/harvest/keeper/keeper.go [hunk 16]
```diff
@@ -178,7 +182,70 @@ func (k Keeper) IterateClaimsByTypeAndDenom(ctx sdk.Context, depositType types.D
...
+	return deposits
+}
+
 // BondDenom returns the bond denom from the staking keeper
 func (k Keeper) BondDenom(ctx sdk.Context) string {
 	return k.stakingKeeper.BondDenom(ctx)
 }
+
+// GetBorrow returns a Borrow from the store for a particular borrower address and borrow denom
+func (k Keeper) GetBorrow(ctx sdk.Context, borrower sdk.AccAddress, denom string) (types.Borrow, bool) {
```
Comment (karzak) [inline x/harvest/keeper/keeper.go:203]: Created #707

File: x/harvest/types/params.go [hunk 17]
```diff
@@ -218,26 +221,97 @@ func (ds DistributionSchedule) GetMultiplier(name MultiplierName) (Multiplier, b
...
+
+// Validate BorrowLimit
+func (bl BorrowLimit) Validate() error {
+	if bl.MaximumLimit.IsNegative() {
+		return fmt.Errorf("maximum limit cannot be negative: %s", bl.MaximumLimit)
+	}
+	if !bl.LoanToValue.IsPositive() {
+		return fmt.Errorf("loan-to-value must be a positive integer: %s", bl.LoanToValue)
+	}
+	if bl.LoanToValue.GT(sdk.NewDec(100)) {
```
Comment (karzak) [inline x/harvest/types/params.go:246]:
```suggestion
	if bl.LoanToValue.GT(sdk.OneDec()) {
```

shouldn't allow under-collateralized loans

File: x/harvest/keeper/borrow.go [hunk 18]
```diff
@@ -0,0 +1,138 @@
...
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/kava-labs/kava/x/harvest/types"
+)
+
+// USDX is the USDX coin's denom
+const USDX = "usdx"
+
+// Borrow funds
+func (k Keeper) Borrow(ctx sdk.Context, borrower sdk.AccAddress, amount sdk.Coin) error {
```
Comment (karzak) [inline x/harvest/keeper/borrow.go:16]: add keeper test for happy-path barrow

File: migrate/v0_11/migrate.go [hunk 19]
```diff
@@ -633,29 +633,34 @@ func MigrateGov(oldGenState v39_1gov.GenesisState) v39_1gov.GenesisState {
...
-		},
-		v0_11harvest.DelegatorDistributionSchedules{v0_11harvest.NewDelegatorDistributionSchedule(
-			v0_11harvest.NewDistributionSchedule(true, "ukava", incentiveGoLiveDate, incentiveEndDate, sdk.NewCoin("hard", sdk.NewInt(1267522)), claimEndDate, v0_11harvest.Multipliers{v0_11harvest.NewMultiplier(v0_11harvest.Small, 1, sdk.MustNewDecFromStr("0.33")), v0_11harvest.NewMultiplier(v0_11harvest.Large, 12, sdk.OneDec())}),
-			time.Hour*24,
-		),
-		},
-	), v0_11harvest.DefaultPreviousBlockTime, v0_11harvest.DefaultDistributionTimes)
-	return harvestGS
-}
+// // MigrateHarvest initializes the harvest genesis state for kava-4
```
Comment (karzak) [inline migrate/v0_11/migrate.go:636]: created #708

File: x/harvest/types/params.go [hunk 20]
```diff
@@ -218,26 +221,97 @@ func (ds DistributionSchedule) GetMultiplier(name MultiplierName) (Multiplier, b
...
+		return fmt.Errorf("loan-to-value cannot be greater than 100: %s", bl.LoanToValue)
+	}
+	return nil
+}
+
+// MoneyMarket is a money market for an individual asset
+type MoneyMarket struct {
```
Comment (karzak) [inline x/harvest/types/params.go:253]: created #709

**Review by karzak — CHANGES_REQUESTED** [review]
Address requested changes before merge

Issues #706, #707, #708, and #709 are non-blocking

File: x/harvest/keeper/borrow.go (hunk 6 above)
Comment (denalimarsh) [reply x/harvest/keeper/borrow.go:67]: Will add `conversion_factor`, which will replace the `math.Pow(10, 6)` here. But the specific USDX if/else condition is to avoid checking the pricefeed for the USDX price when it would return an error as there's no USDX market - so we'll need to add an additional parameter `stablecoin` in order to replace the USDX specific if/else condition.

In this proposal the if/else would remain, but would check for `param.Stablecoin == true` instead of `amount.Denom == "usdx"`. This solution doesn't feel great since BUSD is also a stablecoin but exists in the pricefeed so we can check for price the normal way.

Comment (denalimarsh) [reply x/harvest/keeper/borrow.go:67]: Unless we'll add a pricefeed for USDX? I'm thinking that we shouldn't do this as the protocol naturally assumes that USDX's price is $1.00.

File: x/harvest/types/params.go (hunk 17 above)
Comment (denalimarsh) [reply x/harvest/types/params.go]: https://github.com/Kava-Labs/kava/pull/702/commits/5428519008577591be427331d9501c142a1e77c8

File: x/harvest/keeper/borrow.go (hunk 14 above)
Comment (denalimarsh) [reply x/harvest/keeper/borrow.go]: https://github.com/Kava-Labs/kava/pull/702/commits/5428519008577591be427331d9501c142a1e77c8

File: x/harvest/keeper/borrow.go (hunk 13 above)
Comment (denalimarsh) [reply x/harvest/keeper/borrow.go]: https://github.com/Kava-Labs/kava/pull/702/commits/5428519008577591be427331d9501c142a1e77c8

File: x/harvest/keeper/borrow.go (hunk 12 above)
Comment (denalimarsh) [reply x/harvest/keeper/borrow.go]: Good catch, resolved in https://github.com/Kava-Labs/kava/pull/702/commits/5428519008577591be427331d9501c142a1e77c8

File: x/harvest/keeper/borrow.go (hunk 11 above)
Comment (denalimarsh) [reply x/harvest/keeper/borrow.go]: https://github.com/Kava-Labs/kava/pull/702/commits/5428519008577591be427331d9501c142a1e77c8

File: x/harvest/keeper/borrow.go (hunk 10 above)
Comment (denalimarsh) [reply x/harvest/keeper/borrow.go]: Moved as requested in https://github.com/Kava-Labs/kava/pull/702/commits/5428519008577591be427331d9501c142a1e77c8.

Argument for moving it back up: if the user has no deposits then it'll be impossible to successfully borrow an asset. So shouldn't we throw this error once as soon as we know the user has no deposits?

File: x/harvest/keeper/borrow.go (hunk 6 above)
Comment (denalimarsh) [reply x/harvest/keeper/borrow.go]: Addressed the `conversation_factor` part of this in https://github.com/Kava-Labs/kava/pull/702/commits/277c4b1936981182fc09ae0c6ea0973d9cbbca8a.

File: x/harvest/types/params.go (hunk 7 above)
Comment (denalimarsh) [reply x/harvest/types/params.go]: Included in https://github.com/Kava-Labs/kava/pull/702/commits/277c4b1936981182fc09ae0c6ea0973d9cbbca8a

File: x/harvest/keeper/borrow.go (hunk 8 above)
Comment (denalimarsh) [reply x/harvest/keeper/borrow.go]: Deleted in https://github.com/Kava-Labs/kava/pull/702/commits/277c4b1936981182fc09ae0c6ea0973d9cbbca8a

File: x/harvest/keeper/borrow.go (hunk 9 above)
Comment (denalimarsh) [reply x/harvest/keeper/borrow.go]: Refactored this function to `ValidateBorrow` in https://github.com/Kava-Labs/kava/pull/702/commits/277c4b1936981182fc09ae0c6ea0973d9cbbca8a.

Also moved price fetching to a helper function. It's a bit cleaner, but still not convinced it's the best way as we're still double-fetching the borrow asset's MoneyMarket.

File: x/harvest/keeper/borrow.go (hunk 6 above)
Comment (karzak) [reply x/harvest/keeper/borrow.go]: We'll add a pricefeed for USDX

File: x/harvest/keeper/borrow.go (hunk 10 above)
Comment (karzak) [reply x/harvest/keeper/borrow.go]: Yeah, that's what I meant. This should be the first check and we can short-circuit error if the user has no deposits. Sorry if unclear.

File: x/harvest/keeper/borrow.go (hunk 18 above)
Comment (denalimarsh) [reply x/harvest/keeper/borrow.go]: Added in https://github.com/Kava-Labs/kava/pull/702/commits/25a9015bd845e9b260444d1578b5e928667a0031, https://github.com/Kava-Labs/kava/pull/702/commits/390116c0f1e3e480feb21ecf6257cc45b6e2a7c0

**Review by karzak — APPROVED** [review]
LGTM

Left a few nits. Also, please update `harvest/types` tests to fix the build before merging (you can just use `types.DefaultMoneyMarkets` to fix the issues).

File: x/harvest/keeper/borrow_test.go [hunk 21]
```diff
@@ -0,0 +1,101 @@
...
+		suite.Run(tc.name, func() {
+			// create new app with one funded account
+
+			// Initialize test app and set context
+			tApp := app.NewTestApp()
+			ctx := tApp.NewContext(true, abci.Header{Height: 1, Time: tmtime.Now()})
+			authGS := app.NewAuthGenState(
+				[]sdk.AccAddress{tc.args.borrower},
+				[]sdk.Coins{sdk.NewCoins(sdk.NewCoin("ukava", sdk.NewInt(100)))})
+			loanToValue, _ := sdk.NewDecFromStr("0.6")
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/harvest/keeper/borrow_test.go]: https://github.com/Kava-Labs/kava/pull/702/commits/295405be8dfa3fe2497b828c79ac8b6bda4fb1f7

File: x/harvest/keeper/keeper.go [hunk 22]
```diff
@@ -178,7 +182,70 @@ func (k Keeper) IterateClaimsByTypeAndDenom(ctx sdk.Context, depositType types.D
...
+	for ; iterator.Valid(); iterator.Next() {
+		var borrow types.Borrow
+		k.cdc.MustUnmarshalBinaryBare(iterator.Value(), &borrow)
+		if cb(borrow) {
+			break
+		}
+	}
+}
+
+// // GetBorrowsByUser gets all borrows for an individual user
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/harvest/keeper/keeper.go]: https://github.com/Kava-Labs/kava/pull/702/commits/295405be8dfa3fe2497b828c79ac8b6bda4fb1f7

File: x/harvest/client/cli/query.go [hunk 23]
```diff
@@ -250,3 +252,61 @@ func queryClaimsCmd(queryRoute string, cdc *codec.Codec) *cobra.Command {
...
+			if err := cdc.UnmarshalJSON(res, &borrows); err != nil {
+				return fmt.Errorf("failed to unmarshal borrows: %w", err)
+			}
+			return cliCtx.PrintOutput(borrows)
+		},
+	}
+	cmd.Flags().Int(flags.FlagPage, 1, "pagination page to query for")
+	cmd.Flags().Int(flags.FlagLimit, 100, "pagination limit (max 100)")
+	cmd.Flags().String(flagOwner, "", "(optional) filter for borrows by owner address")
+	cmd.Flags().String(flagBorrowDenom, "", "(optional) filter for borrows by denom")
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply x/harvest/client/cli/query.go]: https://github.com/Kava-Labs/kava/pull/702/commits/295405be8dfa3fe2497b828c79ac8b6bda4fb1f7
